src/sklearn_evaluation/plot/_example.py

```diff
@@ -0,0 +1,237 @@
+"""
+Example implemention of ``AbstractPlot``, and ``ComposedAbstractPlot``. This is
+intended to guide developers add new plots, and not intended for new users.
+
+Useful links:
+
+.. plot:: documentation https://matplotlib.org/3.1.3/devel/plot_directive.html
+"""
+from collections import Counter
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn_evaluation import __version__
+from sklearn_evaluation.plot.plot import AbstractComposedPlot, AbstractPlot
+
+
+class MyBar(AbstractPlot):
+    """
+    Bar plot. This is an internal plot targeted for developers, not intended for
+    end-users.
+
+    Parameters
+    ----------
+    count : dict
+        A dictionary whose keys are labels and values are counts
+
+    color : string, default=None
+        Color for the bars, must be a valid matplotlib color
+
+    name : string, default=None
+        A value to identify this plot
+
+    Notes
+    -----
+    .. versionadded:: 0.9
+
+
+    Examples
+    --------
+
+    Create plot:
+
+    .. plot::
+
+        from sklearn_evaluation.plot._example import MyBar
+        MyBar.from_raw_data(["banana", "banana", "apple", "pineapple", "apple"],
+                                color="lightblue")
+
+    Compare plots:
+
+
+    .. plot::
+
+        from sklearn_evaluation.plot._example import MyBar
+        one = MyBar.from_raw_data(["banana", "banana", "apple", "pineapple", "apple"])
+        another = MyBar.from_raw_data(["banana", "apple",  "pineapple"])
+        one + another
+
+    """
+
+    def __init__(self, count, *, color=None, name=None):
+        self.count = count
+        self.color = color
+        self.name = name
+
+    # ax must be optional
+    def plot(self, ax=None):
+        """Create the plot
+
+        Parameters
+        -----------
+        ax : matplotlib.Axes
+            An Axes object to add the plot to
+        """
+        if ax is None:
+            _, ax = plt.subplots()
+
+        ax.bar(
+            list(range(len(self.count))), list(self.count.values()), color=self.color
+        )
+
+        # properly label axis and ticks
+        ax.set_xticks(list(range(len(self.count))), self.count.keys())
+
+        ax.set_ylabel("Count")
+        label = "" if self.name is None else f" ({self.name})"
+        ax.set_title("Bar plot" + label)
+
+        # remember to set these attributes
+        self.ax_ = ax
+        self.figure_ = ax.figure
+
+        # remember to return self
+        return self
+
+    @classmethod
+    def from_raw_data(cls, things_to_count, *, color=None, name=None):
```

Comment:
Shouldn't this method take `ax` as optional parameter as well, in case user wants to pas a custom figure?

Reply:
since we've moved the plotting logic to the `.plot()` method, it'd be `.plot()` what takes `ax`.

e.g.

```python
cm = ConfusionMatrix(...)
cm.plot(ax=ax)
```

But good point, I realized I never thought about this, but we're good!